homeassistant/components/icloud.py

```diff
@@ -0,0 +1,946 @@
+
+CONF_EVENTS = 'events'
+DEFAULT_EVENTS = False
+
+CONF_COOKIEDIRECTORY = 'cookiedirectory'
+
+# entity attributes
+ATTR_ACCOUNTNAME = 'accountname'
+ATTR_INTERVAL = 'interval'
+ATTR_DEVICENAME = 'devicename'
+ATTR_BATTERY = 'battery'
+ATTR_DISTANCE = 'distance'
+ATTR_STARTTIME = 'start_time'
+ATTR_ENDTIME = 'end_time'
+ATTR_DURATION = 'duration'
+ATTR_REMAINING = 'remaining_time'
+ATTR_DEVICESTATUS = 'device_status'
+ATTR_LOWPOWERMODE = 'low_power_mode'
+ATTR_BATTERYSTATUS = 'battery_status'
+ATTR_LOCATION = 'location'
+ATTR_FRIENDLY_NAME = 'friendly_name'
+ATTR_GOOGLE_MAPS_TRAVEL_TIME = 'google_maps_travel_time'
+ATTR_GOOGLE_MAPS_TRAVEL_TIME_DURATION = 'gmtt_duration'
+ATTR_GOOGLE_MAPS_TRAVEL_TIME_ORIGIN = 'gmtt_origin'
+
+TYPE_CURRENT = 'currentevent'
+TYPE_NEXT = 'nextevent'
+
+ICLOUDTRACKERS = {}
+
+DOMAIN = 'icloud'
+DOMAIN2 = 'idevice'
+DOMAIN3 = 'ievent'
+
+ENTITY_ID_FORMAT_ICLOUD = DOMAIN + '.{}'
+ENTITY_ID_FORMAT_DEVICE = DOMAIN2 + '.{}'
+ENTITY_ID_FORMAT_EVENT = DOMAIN3 + '.{}'
+
+DEVICESTATUSSET = ['features', 'maxMsgChar', 'darkWake', 'fmlyShare',
+                   'deviceStatus', 'remoteLock', 'activationLocked',
+                   'deviceClass', 'id', 'deviceModel', 'rawDeviceModel',
+                   'passcodeLength', 'canWipeAfterLock', 'trackingInfo',
+                   'location', 'msg', 'batteryLevel', 'remoteWipe',
+                   'thisDevice', 'snd', 'prsId', 'wipeInProgress',
+                   'lowPowerMode', 'lostModeEnabled', 'isLocating',
+                   'lostModeCapable', 'mesg', 'name', 'batteryStatus',
+                   'lockedTimestamp', 'lostTimestamp', 'locationCapable',
+                   'deviceDisplayName', 'lostDevice', 'deviceColor',
+                   'wipedTimestamp', 'modelDisplayName', 'locationEnabled',
+                   'isMac', 'locFoundEnabled']
+
+
+def setup(hass, config):  # pylint: disable=too-many-locals,too-many-branches
+    """Set up the iCloud Scanner."""
+    if config.get(DOMAIN) is None:
+        return False
+
+    for account, account_config in config[DOMAIN].items():
+        if not isinstance(account_config, dict):
+            _LOGGER.error("Missing configuration data for account %s", account)
+            continue
+        if CONF_USERNAME not in account_config:
+            _LOGGER.error("Missing username for account %s", account)
+            continue
+        if CONF_PASSWORD not in account_config:
+            _LOGGER.error("Missing password for account %s", account)
+            continue
+
+        # Get the username and password from the configuration
+        username = account_config.get(CONF_USERNAME)
+        password = account_config.get(CONF_PASSWORD)
+        cookiedirectory = account_config.get(CONF_COOKIEDIRECTORY, None)
+        getevents = account_config.get(CONF_EVENTS, DEFAULT_EVENTS)
+
+        ignored_devices = []
+        if 'ignored_devices' in account_config:
+            ignored_dev = account_config.get('ignored_devices')
```

Comment:
`.get('ignored_devices', [])` and then you dont need the "if"
Since this is from config, it might be good to create a constant that you share between the voluptuous schema and this get